test/solver/model_getters.jl

```diff
@@ -0,0 +1,43 @@
+import ClimaAtmos as CA
+
+@testset "Model config" begin
+    config = CA.AtmosConfig(Dict("config" => "box"))
+    parsed_args = config.parsed_args
+    @test CA.get_model_config(parsed_args) isa CA.BoxModel
+
+    config = CA.AtmosConfig(Dict("config" => "plane"))
+    parsed_args = config.parsed_args
+    @test CA.get_model_config(parsed_args) isa CA.PlaneModel
+end
```

Comment:
Model config is another place where we can refactor: if we use Grids, we immediately know if we are on a Plane or a Box